### ISSUE TITLE
Missing uiEditableComboboxClear, uiEditableComboboxDelete, uiEditableComboboxInsertAt, uiEditableComboboxNumItems

### ISSUE THREAD

**Issue by AndyObtiva** [libui-ng/libui-ng#299]
Combo box has the following methods that allow querying items, deleting items, and re-inserting items in any order:
Missing uiComboboxClear, uiComboboxDelete, uiComboboxInsertAt, uiComboboxNumItems

uiEditableCombobox does not have them, which prevents me from being able to clear and replace items for an existing uiEditableComboBox after appending items to it.

I assume implementing those missing methods is not much effort given they are similar to those of uiCombobox

**Comment by rubyFeedback** [comment]
Perhaps we can have a patch-centric fork of libui-ng, to track those changes and apply them (specifically), in case there is code / pull request that adds such functionality.

**Comment by petabyt** [comment]
I was just talking about making something like that yesterday

**Comment by cody271** [comment]
> Perhaps we can have a patch-centric fork of libui-ng, to track those changes and apply them (specifically), in case there is code / pull request that adds such functionality.
>
GTK used to have something like this called `libegg`, a repo for its patchset of various "extras" / controls:
https://mail.gnome.org/archives/gtk-app-devel-list/2007-May/msg00067.html

**Comment by cody271** [comment]
> I assume implementing those missing methods is not much effort given they are similar to those of uiCombobox
>
Yes these methods look to be entirely missing from the editable control. Definitely need to be added.

There are a few subtle bugs in both combobox controls I need to fix first, like debounce for example:
```patch
diff --git a/darwin/combobox.m b/darwin/combobox.m
index 1c0aaf4f..75053528 100644
--- a/darwin/combobox.m
+++ b/darwin/combobox.m
@@ -10,6 +10,7 @@ struct uiCombobox {
 	NSArrayController *pbac;
 	void (*onSelected)(uiCombobox *, void *);
 	void *onSelectedData;
+	int selected;
 };
 
 @interface uiprivCombobox : NSPopUpButton {
@@ -44,7 +45,13 @@ struct uiCombobox {
 - (IBAction)onSelected:(id)sender
 {
 	uiCombobox *c = self->combobox;
+	NSInteger index;
 
+	index = [c->pb indexOfSelectedItem];
+	if (index == c->selected)
+		return;
+
+	c->selected = index;
 	(*(c->onSelected))(c, c->onSelectedData);
 }
 
@@ -114,11 +121,12 @@ int uiComboboxNumItems(uiCombobox *c)
 
 int uiComboboxSelected(uiCombobox *c)
 {
-	return [c->pb indexOfSelectedItem];
+	return c->selected;
 }
 
 void uiComboboxSetSelected(uiCombobox *c, int n)
 {
+	c->selected = n;
 	[c->pb selectItemAtIndex:n];
 }
```

**Comment by cody271** [comment]
New QA page for comboboxes to expose these bugs, which should also be fairly easy to port to tabs as well.

<img width="536" alt="qa_combobox" src="https://github.com/user-attachments/assets/2494c5b9-adad-4675-875c-76512934dd39">
